content/post/practical_ml_2.md

```diff
@@ -0,0 +1,55 @@
+ 
+* *Seasonality*: Events around Christmas time look very different from events in the summer.
+* *Marketing efforts*: Different types of campaigns will attract different types of users, with different behaviors. 
+* *Changes in Regulatory landscape*: [GDPR](https://en.wikipedia.org/wiki/General_Data_Protection_Regulation?oldformat=true) is a prime example of this. What happens when we have to delete user data that our model has been leveraging?
+* *Product launches*: It is very common for tech startups to pivot unexpectedly. This wreaks havoc for our models.
+ 
+In order to combat the reality of a changing data distribution, the Data Scientist must retrain her model at regular intervals, perform feature engineering as needed and potentially change the entire algorithm that underlies her model. While we can strive to minimize our model's generalization error, we must recognize that our job is not complete just because we hit deploy.
+ 
+## Technical considerations
+ 
+Unless the Data Scientist's model is deployed on production, it usually runs on a Data Warehouse and likely does not interact with the application. This may result in the Data Scientist cutting some corners; it is not uncommon to sacrifice model optimization for increased accuracy. After all, if a model is not used in production then stakeholders don't need real-time results, making optimization a "nice to have". In addition, Data Scientists are often not trained in software engineering best practices. A Data Scientist’s model is usually a script having minimal interaction with a larger application. The range of a model’s inputs and outputs are much more narrow and easily controlled.
+ 
+However, consider our spam filter. Every time a user sends an event we must classify it as spam or not. Our filter is part of a series of checkpoints between when the user hits *send* and when the email gets delivered in their recipient's inbox. That is very real-time. In this paradigm, optimization and efficiency are critical. If we trained a heavy model, however accurate, that slowed down our email deliverability, it would never see the light of day. Moreover, since this classifier interacts with the application, it must in some way be integrated with the greater application infrastructure. This means more scrutiny to testing, exception handling, logging and other software engineering best practices. 
+ 
+In this context, the Data Scientist quickly realizes that her statistical model is but a very small part in a large web of interconnected processes. As a result, it may be prudent to spend less time tweaking a model to achieve some theoretical guarantee and more time understanding how it will fit in the bigger picture.
+ 
+## Show and tell
+ 
+I am a strong proponent of frequent presentations, discussions and knowledge sharing. This is especially important when working on a large, complex Machine Learning model. Presenting one's work has the following benefits:
```

Comment:
If you buy my view on tying together these sections on the theme of being robust to change, this intro would benefit from a bit of reframing along those lines.

Reply:
Yeah makes sense.